### PR TITLE
Remove `nmdc_schema/nmdc_schema_merged.yaml` along with all references to it

### PR DESCRIPTION
**Pull request by turbomam**
see
- https://github.com/microbiomedata/nmdc-schema/issues/2095

### CODE REVIEW

**Comment by eecavanna** [comment]
I recommend removing this doctest (two lines of a docstring), since the two schemas being compared are the same.

https://github.com/microbiomedata/nmdc-schema/pull/2099/files#diff-c97c98fe37b8d9f073ff92b10894c3136526b840bb4d96c86a4794815cbd06d7R119-R120

On a side note, I'm surprised this doctest didn't fail here in the PR, when GHA ran the tests.

**Comment by eecavanna** [comment]
I like this PR title.

File: nmdc_schema/nmdc_data.py [hunk 1]
```diff
@@ -103,7 +103,7 @@ def get_nmdc_jsonschema_bytesIO(variant: Optional[SchemaVariantIdentifier] = Non
 
 
 def get_nmdc_jsonschema_bytes(variant: Optional[SchemaVariantIdentifier] = None) -> bytes:
-    """Reruns the nmdc.schema.json file as bytes.
+    """Retruns the nmdc.schema.json file as bytes.
```
Comment (eecavanna) [inline nmdc_schema/nmdc_data.py:106]: Fix typo

Comment (eecavanna) [reply nmdc_schema/nmdc_data.py:106]:
```suggestion
    """Returns the nmdc.schema.json file as bytes.
```

**Review by aclum — APPROVED** [review]

**Comment by eecavanna** [comment]
I met briefly with @turbomam and it turned out I had been thinking of an unrelated conversation when I was reviewing this PR. That led to me requesting (above) that he remove a doctest. As he and I discussed, I will make a commit on this branch that restores that doctest.

**Review by eecavanna — APPROVED** [review]

**Review by mslarae13 — COMMENTED** [review]
No suggestions to me & again I'll defer to @eecavanna 
That said I did have a question on the issue for my own clarification.

**Comment by turbomam** [comment]
> That said I did have a question on the issue for my own clarification.

Thanks. Did [this](https://github.com/microbiomedata/nmdc-schema/issues/2095#issuecomment-2214203188) answer your questions?

**Comment by ssarrafan** [comment]
> > That said I did have a question on the issue for my own clarification.
> 
> Thanks. Did [this](https://github.com/microbiomedata/nmdc-schema/issues/2095#issuecomment-2214203188) answer your questions?

@mslarae13 I'm assuming your question was answered in the source issue correct?  Can we close this?  It looks like it was reviewed by Alicia and Eric too.

**Review by naglepuff — APPROVED** [review]

**Pull request merged**